src/cart/crt_iv.c

```diff
@@ -1086,8 +1086,8 @@ handle_ivfetch_response(const struct crt_cb_info *cb_info)
 	}
 
 	/* In case of a failure, call on_refresh with NULL iv_value */
-	iv_ops->ivo_on_refresh(ivns, &input->ifi_key, 0, /* TODO: iv_ver */
-			       iv_value, false, rc, iv_info->ifc_user_priv);
+	rc = iv_ops->ivo_on_refresh(ivns, &input->ifi_key, 0, /* TODO: iv_ver */
```

Comment:
i dont think this is correct. rc is only used in this function later to determine the order of finalizing pending events on lines 1113-1120, and it should be using the rc from the fetch, not the success/failure of refreshing the value locally.
it functionally will do the right thing today, because of how you changed iv_on_refresh() implementation to return the same rc, but I also don't think that is a proper way, as it should return success instead.

Reply:
yes, rc will be used later indeed.  But if on_refresh failed, which also validate the fetch value actually,  then the return value should be updated here.   So there are two cases

if fetch failed, i.e. the original rc is not 0, then ivo_on_refresh will return the original rc anyway(see above change),  

if fetch succeeds, but ivo_on_refresh failed due to local validation or refresh, then rc should be updated and used later, then if the local engine will have to do IV remote fetch again, if it want to get the update value.